src/main/java/org/jawk/intermediate/AwkTuples.java

```diff
@@ -1645,7 +1643,7 @@ public static String toOpcodeString(int opcode) {
 		Field[] fields = c.getDeclaredFields();
 		try {
 			for (Field field : fields) {
-				if ((field.getModifiers() & Modifier.STATIC) > 0 && field.getType() == Integer.TYPE && field.getInt(null) == opcode) {
+				if ((field.getModifiers() & Modifier.STATIC) > 0 && field.getType() == Long.TYPE && field.getLong(null) == opcode) {

```

Comment:
Totally agree with you. Problem is: the original code uses the same methods/functions to work on opcodes and values. So, when I started changing things to have *long* variables instead of *int*, it also broke the opcode thing, that I had to update.
To be perfect, we would need to separate opcode-related functions to variable-related functions. But that needs deeper refactoring and thinking with a much higher probably of introducing bugs difficult to detect. That's why I went for the simpler option and use *long* everywhere where Eclipse would tell me it's needed...

Reply:
I don't think I understand. From the looks of it this function (`toOpcodeString`) is broken now, since it's trying to look for `long` variables where in fact all of the opcodes are still `int`s.